src/main/java/io/vlingo/schemata/model/ContextEntity.java

```diff
@@ -7,89 +7,55 @@
 
 package io.vlingo.schemata.model;
 
-import java.util.function.BiConsumer;
-
-import io.vlingo.lattice.model.sourcing.EventSourced;
+import io.vlingo.common.Completes;
+import io.vlingo.lattice.model.object.ObjectEntity;
 import io.vlingo.schemata.model.Events.ContextDefined;
 import io.vlingo.schemata.model.Events.ContextDescribed;
 import io.vlingo.schemata.model.Events.ContextRenamed;
 import io.vlingo.schemata.model.Id.ContextId;
 
-public class ContextEntity extends EventSourced implements Context {
-  private State state;
 
-  public ContextEntity(final ContextId contextId) {
-    this.state = new State(contextId);
+public class ContextEntity extends ObjectEntity<ContextState> implements Context {
+  private ContextState state;
+
+  public ContextEntity() {
+    this.state = new ContextState();
   }
 
   @Override
-  public void defineWith(final String name, final String description) {
+  public Completes<ContextState> defineWith(final ContextId contextId, final String name, final String description) {
     assert (name != null && !name.isEmpty());
     assert (description != null && !description.isEmpty());
-    apply(ContextDefined.with(state.contextId, name, description));
+    apply(state.define(contextId, name, description), ContextDefined.with(contextId, name, description), () -> state);
```

Comment:
why we need that `() -> state` ?

Reply:
Also note that if you wanted to instantiate a new entity with its initial state you can have that auto-persisted by overriding `withNewState()` and return the initial state and any applicable `List<Source<C>>` with it.

See:
https://github.com/vlingo/vlingo-lattice/blob/04cd492128b52af5c51b404432998cbbefd21be1/src/main/java/io/vlingo/lattice/model/object/ObjectEntity.java#L62

And:
https://github.com/vlingo/vlingo-lattice/blob/04cd492128b52af5c51b404432998cbbefd21be1/src/main/java/io/vlingo/lattice/model/object/ObjectEntity.java#L187

Reply:
@VaughnVernon why the parameter `List<Source<C>>` is called the way when we return events, or as I saw just one event? Also I don't have any events to return, just an empty list in that case.